web/components/roadmap/MiniRoadmap.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { ReactElement, useCallback, useEffect, useState } from "react";
+import { useRoadmap } from "@/lib/data-hooks/useRoadmap";
+import { VerticalStepIndicator } from "@/components/njwds-extended/VerticalStepIndicator";
+import { MiniRoadmapTask } from "@/components/roadmap/MiniRoadmapTask";
+import { Icon } from "@/components/njwds/Icon";
+import analytics from "@/lib/utils/analytics";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { isStepCompleted } from "@/lib/utils/helpers";
+import { SectionType } from "@/lib/types/types";
+import { SectionDefaults } from "@/display-content/roadmap/RoadmapDefaults";
+import { Accordion, AccordionSummary, AccordionDetails } from "@mui/material";
+import { getSectionNames } from "@/lib/utils/helpers";
+
+interface Props {
+  activeTaskId: string;
+  onTaskClick?: () => void;
+}
+
+export const MiniRoadmap = (props: Props): ReactElement => {
+  const { roadmap } = useRoadmap();
+  const getActiveStepNumber = useCallback(
+    () =>
+      roadmap?.steps.find((step) => step.tasks.map((it) => it.id).includes(props.activeTaskId))?.step_number,
+    [props.activeTaskId, roadmap?.steps]
+  );
+  const [activeStepNumber, setActiveStepNumber] = useState<number | undefined>(getActiveStepNumber());
+  const [openSteps, setOpenSteps] = useState<number[]>([]);
+  const { userData, update } = useUserData();
+  const roadmapExists = !!roadmap;
+  const roadmapSections = getSectionNames(roadmap);
+
+  useEffect(() => {
+    const activeStep = getActiveStepNumber();
+    setActiveStepNumber(activeStep);
+    setOpenSteps(activeStep ? [activeStep] : []);
+  }, [props.activeTaskId, roadmapExists, getActiveStepNumber]);
+
+  const toggleStep = (stepNumber: number): void => {
+    analytics.event.task_mini_roadmap_step.click.expand_contract();
+    if (openSteps.includes(stepNumber)) {
+      setOpenSteps(openSteps.filter((number) => number !== stepNumber));
+    } else {
+      setOpenSteps([...openSteps, stepNumber]);
+    }
+  };
+
+  const handleAccordionStateChange = async (
```

Comment:
this handler code is duplicated in RoadmapBySections, as is a lot of the HTML below. Their main difference is the contents of the AccordianDetails, it seems? This seems like an opportunity for refactoring some kind of Roadmap Accordian component so that this handler and all that code could only be written once, and used in both places

Reply:
Refactored to a utility function called by both RoadmapBySections and MiniRoadmap.

Reply:
I think I was suggesting a component more than a utility function (so that the duplicated JSX can also be refactored) - if you think that makes sense, of course

Reply:
Sorry, should have been explicit, the utility function is just for the handler. Will address the JSX portion.

Reply:
Though the JSX are similar in some cases for both RoadmapBySections and MiniRoadmap, there are some key differences concerning their child elements and class names / styles. I'd suggest we leave them both as is.